MSAL/test/app/mac/MSALCacheViewController.m

```diff
@@ -26,16 +26,305 @@
+{
+    if (accessToken)
+    {
+        accessToken.expiresOn = [NSDate dateWithTimeIntervalSinceNow:-1.0];
+        [self.tokenCache saveCredential:accessToken.tokenCacheItem context:nil error:nil];
+        [self loadCache];
+    }
+}
+
+- (void)deleteAllEntriesForAccount:(MSIDAccount *)account
+{
+    if (account)
+    {
+        [self.defaultAccessor clearCacheForAccount:account.accountIdentifier
+                                         authority:nil
+                                          clientId:nil
+                                          familyId:nil
+                                           context:nil
+                                             error:nil];
+        
+        [self loadCache];
+    }
+}
+
+- (void)deleteAppMetadata:(MSIDAppMetadataCacheItem *)appMetadata
+{
+    if (appMetadata)
+    {
+        [self.tokenCache removeAppMetadata:appMetadata context:nil error:nil];
+        [self loadCache];
+    }
+}
+
+- (void)deleteToken:(MSIDBaseToken *)token
+{
+    if (token)
+    {
+        switch (token.credentialType)
+        {
+            case MSIDRefreshTokenType:
+            {
+                [self.defaultAccessor validateAndRemoveRefreshToken:(MSIDRefreshToken *)token
+                                                            context:nil
+                                                              error:nil];
+                break;
+            }
+            default:
+                [self.defaultAccessor removeToken:token context:nil error:nil];
+                break;
+        }
+        
+        [self loadCache];
+    }
+}
+
+#pragma mark Button Actions
+
+- (IBAction)refreshCache:(id)sender
+{
+    [self loadCache];
+}
+
+
+- (IBAction)expireOrInvalidateToken:(id)sender
+{
+    id item = [self.outLineView itemAtRow:[self.outLineView selectedRow]];
+    
+    if ([item isKindOfClass:[MSIDBaseToken class]])
+    {
+        if ([item isKindOfClass:[MSIDRefreshToken class]])
+        {
+            [self invalidateRefreshToken:(MSIDRefreshToken *)item];
+        }
+        else if([item isKindOfClass:[MSIDAccessToken class]])
+        {
+            [self expireAccessToken:(MSIDAccessToken *)item];
+        }
+    }
+}
+
+- (IBAction)deleteItem:(id)sender
+{
+    id item = [self.outLineView itemAtRow:[self.outLineView selectedRow]];
+    if ([item isKindOfClass:[MSIDAccount class]])
+    {
+        [self deleteAllEntriesForAccount:(MSIDAccount *)item];
+    }
+    else if ([item isKindOfClass:[MSIDBaseToken class]])
+    {
+        [self deleteToken:(MSIDBaseToken *)item];
+    }
+    else if ([item isKindOfClass:[MSIDAppMetadataCacheItem class]])
+    {
+        [self deleteAppMetadata:(MSIDAppMetadataCacheItem *)item];
+    }
+}
+
+#pragma mark NSOutlineView Data Source Test Methods
+
+- (NSInteger)outlineView:(NSOutlineView *)outlineView numberOfChildrenOfItem:(id)item
+{
+    return !item ? [[self.cacheDict allKeys] count] : [[self.cacheDict objectForKey:item] count];
```

Comment:
nit: will it be more natural if we do it like `item? [[self.cacheDict objectForKey:item] count] : [[self.cacheDict allKeys] count]` :)

Reply:
Addressed.